nixos/modules/virtualisation/qcow.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.qcow;
+in
+{
+  options = {
+    virtualisation.qcow = {
+      baseImageSize = mkOption {
+        type = with types; either (enum [ "auto" ]) ints.positive;
+        default = "auto";
+        example = 8192;
+        description = ''
+          The base size of the QCOW2 image in MiB.
+        '';
+      };
+      additionalImageSpace = mkOption {
+        type = types.int;
+        default = 512;
+        example = 1024;
+        description = ''
+          Additional disk space in MiB to be added to the image if <literal>virtualisation.qcow.baseImageSize</literal> is set to <literal>auto</literal>.
+        '';
+      };
+      imageBootPartitionSize = mkOption {
+        type = types.int.positive;
+        default = 256;
+        example = 512;
+        description = ''
+          The size of the boot partition in MiB, only takes effect if <literal>virtualisation.qcow.partitionTableType</literal> is set to <literal>efi</literal> or <literalhybrid</literal>.
+        '';
+      };
+      imagePartitionTableType = mkOption {
+        type = types.enum [ "legacy" "efi" "legacy+gpt" "hybrid" "none" ];
+        default = "legacy";
+        example = "efi";
+        description = ''
+          Type of partition table to use for the image, available options are <literal>legacy</literal>, <literal>efi</literal>, <literal>legacy+gpt</literal>, <literal>hybrid</literal> or <literal>none</literal>. For additional information about each option: <link>https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix#L32</link>.
+        '';
+      };
+    };
+  };
+
+  config = {
+    system.build.qcow = import ../../lib/make-disk-image.nix {
```

Comment:
`system.build` was a mistake. It's a mess of undocumented quasi-options. I'd like this to be a proper option with `readOnly = true`. How about `virtualisation.image.file`?

(`image`: see other comment)